### DISCUSSION TITLE
Should `gcp-pub-sub` work with emulated pubsub `PUBSUB_EMULATOR_HOST`

### DISCUSSION THREAD

**Discussion by lskiva-r7** [kedacore/keda#5638] [Q&A / Need Help]
Another pod is using this image:
google/cloud-sdk

We run this on that container in order to emulate pubsub:
```
command: ["gcloud", "beta", "emulators", "pubsub", "start", "--project=test-project", "--host-port=0.0.0.0:8085"]
```

We then run our service using the `gcp-pubsub` trigger.
We do pass the `PUBSUB_EMULATOR_HOST` with the address of the other pod as an environment variable for the command.

We have a test that publishes pubsub to the same subscription using the `PUBSUB_EMULATOR_HOST` during a test.
But the trigger doesn't seem to work when doing that.

Does the `PUBSUB_EMULATOR_HOST` work here?

**Comment by zroubalik** [comment]
Where do you set this env value?

**Reply by lskiva-r7** [reply]
It's defined as `envFrom` on the container
